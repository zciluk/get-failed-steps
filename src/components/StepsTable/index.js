import React from "react";
import { Table, Popup, Label } from "semantic-ui-react";
import _ from "lodash";

const StepsTable = ({ retrievedData, specsData }) => {
  return (
    <Table collapsing color="teal">
      <Table.Header>
        <Table.Row>
          <Table.HeaderCell>Name of test</Table.HeaderCell>
          <Table.HeaderCell>Number of Fails</Table.HeaderCell>
        </Table.Row>
      </Table.Header>

      <Table.Body>
        {_.sortBy(retrievedData, ["fail"])
          .reverse()
          .map(rowData => (
            <Table.Row key={rowData.test}>
              <Table.Cell>
                <Popup
                  trigger={<Label basic>{rowData.test}</Label>}
                  content={specsData
                    .map(elem => {
                      if (elem[0] === rowData.test) return elem[1];
                      return "";
                    })
                    .filter((item, i, ar) => ar.indexOf(item) === i)}
                  inverted
                />
              </Table.Cell>
              <Table.Cell>{rowData.fail}</Table.Cell>
            </Table.Row>
          ))}
      </Table.Body>
    </Table>
  );
};

export default StepsTable;
