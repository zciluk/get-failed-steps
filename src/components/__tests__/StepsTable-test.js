import React from "react";
import renderer from "react-test-renderer";
import StepsTable from "../StepsTable";
const mockData = ["Step 1", "Step2", "Step other", "Step4", "Step 1", "Step 1"];
test("should render table with mocked data", () => {
  const component = renderer.create(<StepsTable displayedData={mockData} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
