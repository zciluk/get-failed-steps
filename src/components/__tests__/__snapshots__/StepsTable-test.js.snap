// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render table with mocked data 1`] = `
<table
  className="ui teal collapsing table"
>
  <thead
    className=""
  >
    <tr
      className=""
    >
      <th
        className=""
      >
        Name of test
      </th>
      <th
        className=""
      >
        Number of Fails
      </th>
    </tr>
  </thead>
  <tbody
    className=""
  >
    <tr
      className=""
    >
      <td
        className=""
      >
        Step 1
      </td>
      <td
        className=""
      >
        3
      </td>
    </tr>
    <tr
      className=""
    >
      <td
        className=""
      >
        Step2
      </td>
      <td
        className=""
      >
        1
      </td>
    </tr>
    <tr
      className=""
    >
      <td
        className=""
      >
        Step other
      </td>
      <td
        className=""
      >
        1
      </td>
    </tr>
    <tr
      className=""
    >
      <td
        className=""
      >
        Step4
      </td>
      <td
        className=""
      >
        1
      </td>
    </tr>
  </tbody>
</table>
`;
